ocs_ci/ocs/fiojob.py

```diff
@@ -0,0 +1,228 @@
+    assert pod_data['kind'] == "List"
+    pod_dict = pod_data['items'][0]
+    assert pod_dict['kind'] == "Pod"
+    pod_name = pod_dict['metadata']['name']
+    logger.info(f"Identified pod name of the finished Job: {pod_name}")
+    pod_name = pod_dict['metadata']['name']
+
+    return pod_name
+
+
+def write_data_via_fio(fio_job_file, write_timeout, pvc_size, target_percentage):
+    """
+    Write data via fio Job (specified in ``tf`` tmp file) to reach desired
+    utilization level, and keep this level for ``minimal_time`` seconds.
+    """
+    # unix timestamp before starting the job so that one can check status
+    # prior the fio job run
+    fio_job_start_ts = time.time()
+
+    # deploy the fio Job to the cluster
+    fio_job_file.create()
+
+    # high level description of the problem, reported in case of a job failure
+    # or timeout
+    error_msg = (
+        f"Job fio failed to write {pvc_size} Gi data on OCS backed "
+        f"volume in expected time {write_timeout} seconds."
+        " If the fio pod were still runing"
+        " (see 'last actual status was' in some previous log message),"
+        " this is caused either by"
+        " severe product performance regression"
+        " or by a misconfiguration of the clusterr, ping infra team.")
+    pod_name = wait_for_job_completion(
+        fio_job_file.project.namespace, write_timeout, error_msg)
+
+    ocp_pod = ocp.OCP(kind="Pod", namespace=fio_job_file.project.namespace)
+    fio_output = ocp_pod.exec_oc_cmd(
+        f"logs {pod_name}", out_yaml_format=False)
+
+    # parse fio output
+    fio_report = fio_to_dict(fio_output)
+
+    logger.info(fio_report)
+
+    # data which will be available to the test via:
+    # fixture_name['result']
+    result = {
+        'fio_job_start': fio_job_start_ts,
+        'fio': fio_report,
+        'pvc_size': pvc_size,
+        'target_p': target_percentage,
+        'namespace': fio_job_file.project.namespace}
+
+    return result
+
+
+def delete_fio_data(fio_job_file, delete_check_func):
+    """
+    Delete fio data by removing the fio job resource, with a wait to
+    make sure date were reclaimed on the ceph level.
+    """
+    # make sure we communicate what is going to happen
+    logger.info(f"going to delete {fio_job_file.name} Job")
+    fio_job_file.delete()
+    logger.info(
+        f"going to wait a bit to make sure that "
+        f"data written by {fio_job_file.name} Job are really deleted")
+
+    check_timeout = 660  # seconds
+    check_sampler = TimeoutSampler(
+        timeout=check_timeout, sleep=30, func=delete_check_func)
+    finished_in_time = check_sampler.wait_for_func_status(result=True)
+    if not finished_in_time:
+        error_msg = (
+            "it seems that the storage space was not reclaimed "
+            f"within {check_timeout} seconds, "
+            "this is most likely a product bug or misconfiguration")
+        logger.error(error_msg)
+        raise Exception(error_msg)
```

Comment:
```suggestion
    assert finished_in_time, (
        "it seems that the storage space was not reclaimed within {check_timeout}"
        " seconds, this is most likely a product bug or misconfiguration"
    )
```

Reply:
I do this on purpose, so that I have this both in the test log and in an exception. While the exception represents the most visible error, directly available from the jenkins report, the log is necessary when checking additional details, and having important errors on both places makes reasoning about the failure easier.

Reply:
ack